onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -621,6 +620,13 @@ void addObjectMethods(py::module& m, Environment& env) {
   py::class_<SessionIOBinding> binding(m, "SessionIOBinding");
   binding
       .def(py::init<InferenceSession*>())
+      .def("bind_input", [](SessionIOBinding* io_binding, const std::string& name, py::object numpy_arr_on_cpu) -> void {
+        OrtValue mlvalue;
+        CreateTensorMLValue(GetAllocator(), name, reinterpret_cast<PyArrayObject*>(numpy_arr_on_cpu.ptr()), &mlvalue);
```

Comment:
I think you'll need to use [CreateGenericMLValue](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/onnxruntime_pybind_mlvalue.cc#L584) here since the input could be either a numpy array or a regular Python array. See how this is invoked [here](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/onnxruntime_pybind_state.cc#L853).

Reply:
Yes, I thought about it. The problem with CreateGenericMLValue is it requires the input_defs. Is there a way iobinding could obtain it?

Reply:
Sync'ed offline.